sdks/python/hatchet_sdk/clients/dispatcher/dispatcher.py

```diff
@@ -95,6 +93,9 @@ async def send_step_action_event(
     async def _try_send_step_action_event(
         self, action: Action, event_type: StepActionEventType, payload: str
     ) -> grpc.aio.UnaryUnaryCall[StepActionEvent, ActionEventResponse]:
+        aio_conn = new_conn(self.config, True)
+        aio_client = DispatcherStub(aio_conn)  # type: ignore[no-untyped-call]
```

Comment:
IMPORTANT: This is one place (I think the only one) where I'm nervous about performance. The issue here is that we can't create an async client like this at the top level without requiring there to be an event loop, so our options are:

1. Throw if there's no event loop (I think this is a bad idea)
2. Create the event loop ourselves (the point of this PR is to _stop_ doing this
3. Do this here (we might be able to cache, I'm not sure though)

Any thoughts on this? I can try timing it and see how it goes